grails-app/services/org/pih/warehouse/importer/InventoryLevelImportDataService.groovy

```diff
@@ -152,13 +155,19 @@ class InventoryLevelImportDataService implements ImportDataService {
     InventoryLevel updateOrCreateInventoryLevel(Location facility, Product product, Map params) {
         // FIXME This should handle duplicate inventory level but I'm not exactly sure
         //  what to do at this point so I'm going to let the user correct duplicates on their own
-        InventoryLevel inventoryLevel = findOrCreateInventoryLevel(facility, product)
+        InventoryLevel inventoryLevel = InventoryLevel.findByInventoryAndProductAndInternalLocationIsNull(facility.inventory, product)
+        if (!inventoryLevel) {
+            inventoryLevel = new InventoryLevel()
+        }
 
+        inventoryLevel.inventory = facility.inventory
+        inventoryLevel.product = product
```

Comment:
suggestion: could move these into the if condition above since they're a part of the findyBy key so we know if we successfully find the inventory level by them we shouldn't need to also assign them again.

Reply:
I actually had a findOrCreate method that handled this better but removed it at the last minute because it was opague. But yes, I agree. I'll try to make this a little more elegant/readable.